sklearn/mixture/tests/test_gaussianmixture.py

```diff
@@ -0,0 +1,557 @@
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'diag':
+        for k, (w, m, c) in enumerate(zip(weights, means,
+                                          covariances['diag'])):
+            X.append(rng.multivariate_normal(m, np.diag(c),
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'tied':
+        for k, (w, m) in enumerate(zip(weights, means)):
+            X.append(rng.multivariate_normal(m, covariances['tied'],
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'full':
+        for k, (w, m, c) in enumerate(zip(weights, means,
+                                          covariances['full'])):
+            X.append(rng.multivariate_normal(m, c, int(np.round(w *
+                                                                n_samples))))
+
+    X = np.vstack(X)
+    return X
+
+
+class RandData(object):
+    n_samples = 500
+    n_components = 2
+    n_features = 2
+    weights = rng.rand(n_components)
+    weights = weights / weights.sum()
+    means = rng.rand(n_components, n_features) * 50
+    covariances = {'spherical': .5 + rng.rand(n_components),
+                   'diag': (.5 + rng.rand(n_components, n_features)) ** 2,
+                   'tied': make_spd_matrix(n_features, random_state=rng),
+                   'full': np.array([make_spd_matrix(n_features,
+                                                     random_state=rng)*.5
+                                     for _ in range(n_components)])}
+    X = dict(zip(COVARIANCE_TYPE, [generate_data(n_samples, n_features,
+                                                 weights, means, covariances,
+                                                 cov_type) for cov_type in
+                                   COVARIANCE_TYPE]))
+    Y = np.hstack([k * np.ones(np.round(w * n_samples)) for k, w in
+                  enumerate(weights)])
+
+
+def test_GaussianMixture_parameters():
+    # test bad parameters
+
+    # n_init should be greater than 0
+    n_init = rng.randint(-10, 1)
+    assert_raise_message(ValueError,
+                         "Invalid value for 'n_init': %d "
+                         "Estimation requires at least one run"
+                         % n_init,
+                         mixture.GaussianMixture, n_init=n_init)
+
+    # covariance_type should be in [spherical, diag, tied, full]
+    covariance_type = 'bad_covariance_type'
+    assert_raise_message(ValueError,
+                         "Invalid value for 'covariance_type': %s "
+                         "'covariance_type' should be in "
+                         "['spherical', 'tied', 'diag', 'full']"
+                         % covariance_type,
+                         mixture.GaussianMixture,
+                         covariance_type=covariance_type)
+
+
+def test__check_X():
```

Comment:
why the double `_`? Better be consistent with the project naming conventions. We don't use double `_` in other test function names.